plugins/editorconfig/src/org/editorconfig/language/codeinsight/inspections/EditorConfigValueUniquenessInspection.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.editorconfig.language.codeinsight.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import org.editorconfig.language.codeinsight.quickfixes.EditorConfigRemoveListValueQuickFix
+import org.editorconfig.language.messages.EditorConfigBundle
+import org.editorconfig.language.psi.EditorConfigOptionValueIdentifier
+import org.editorconfig.language.psi.EditorConfigOptionValueList
+import org.editorconfig.language.psi.EditorConfigVisitor
+import org.editorconfig.language.schema.descriptors.EditorConfigDescriptor
+import org.editorconfig.language.schema.descriptors.impl.EditorConfigListDescriptor
+import org.editorconfig.language.util.associateNotNull
+
+class EditorConfigValueUniquenessInspection : LocalInspectionTool() {
+  override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = object : EditorConfigVisitor() {
+    override fun visitOptionValueList(list: EditorConfigOptionValueList) {
+      val values = list.optionValueIdentifierList
+      val listDescriptor = list.getDescriptor(false) as? EditorConfigListDescriptor ?: return
+      if (listDescriptor.allowRepetitions) return
+
+      val listChildrenDescriptors = mapToDescriptors(values)
+      val message = EditorConfigBundle["inspection.value.uniqueness.message"]
+      for (value in values) {
```

Comment:
.groupBy{ it.getDescriptor() }.filterValues { it.size > 1 }

Reply:
no need for manual hashmap creation (especially using non-standard functions), and re-getting values in each iteration